sql-plugin/src/main/311until340-all/scala/com/nvidia/spark/rapids/shims/GpuSparkPath.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+import scala.language.implicitConversions
+
+/**
+ * A helper type for the implicit conversion between `String` and `SparkPath`.
+ */
+final case class GpuSparkPath(path: String)
```

Comment:
There is no need to put this `GpuSparkPath` case class into shims.

Reply:
Yes. But it is convenient to define the companion object for `GpuSparkPath` in shims, so that we could put the implicit conversions and apply functions in shims. (For Spark340+, we have another apply function `apply(path: SparkPath): GpuSparkPath` define in the companion object.)

Reply:
Do you get any error if moving this case class out of shims? If not, I would prefer to moving it out. Since I saw the code is totally the same and we need to duplicate it again if a new shim is required in the future.

Reply:
We can't define the companion object in a different file I guess.